kasa/device_factory.py

```diff
@@ -129,3 +133,37 @@ def get_device_class_from_info(info: Dict[str, Any]) -> Type[SmartDevice]:
 
         return SmartBulb
     raise UnsupportedDeviceException("Unknown device type: %s" % type_)
+
+
+def get_device_class_from_type_name(device_type: str) -> Optional[Type[SmartDevice]]:
+    """Return the device class from the type name."""
+    supported_device_types: dict[str, Type[SmartDevice]] = {
+        "SMART.TAPOPLUG": TapoPlug,
+        "SMART.KASAPLUG": TapoPlug,
+        "IOT.SMARTPLUGSWITCH": SmartPlug,
+    }
+    return supported_device_types.get(device_type)
+
+
+def get_protocol_from_connection_name(
+    connection_name: str, host: str, credentials: Optional[Credentials] = None
+) -> Optional[TPLinkProtocol]:
+    """Return the protocol from the connection name."""
+    supported_device_protocols: dict[
+        str, Tuple[Type[TPLinkProtocol], Type[TPLinkTransport]]
+    ] = {
+        "IOT.KLAP": (TPLinkIotProtocol, TPLinkKlapTransport),
+        "SMART.AES": (TPLinkSmartProtocol, TPLinkAesTransport),
+        "SMART.KLAP": (TPLinkSmartProtocol, TPlinkKlapTransportV2),
+    }
+    if connection_name not in supported_device_protocols:
+        return None
+
+    protocol_transport_tuple = supported_device_protocols.get(connection_name)
+    transport: TPLinkTransport = protocol_transport_tuple[1](  # type: ignore
+        host, credentials=credentials
+    )
+    protocol: TPLinkProtocol = protocol_transport_tuple[0](  # type: ignore
+        host, credentials=credentials, transport=transport
+    )
```

Comment:
```suggestion
    protocol_class, transport_class = supported_device_protocols.get(connection_name)
    transport: TPLinkTransport = transport_class(  # type: ignore
        host, credentials=credentials
    )
    protocol: TPLinkProtocol = protocol_class(  # type: ignore
        host, credentials=credentials, transport=transport
    )
```
Also, are the type ignores necessary?

Reply:
They were necessary because mypy didn't recognise the null check beforehand.  Now the type:ignore is moved to line  line 162 which is done as per comment